src/messaging/ExchangeContext.cpp

```diff
@@ -142,8 +142,9 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     }
 
     {
-        // ExchangeContext for group are supposed to always be Initiator
-        if (IsGroupExchangeContext() && !IsInitiator())
+        // ExchangeContext for group are supposed to always be Initiator AND not
+        // use TCP as the underlying transport.
+        if (IsGroupExchangeContext() && (!IsInitiator() || GetSessionHandle()->RequiresTCPConnection()))
```

Comment:
Again, there should be no checks for TCP here.

Reply:
Yes, agreed. Removed.